Projektskabelon-master/src/main/java/FunctionLayer/OverlayMaterialCalculator.java

```diff
@@ -54,20 +54,22 @@ public static Material screwSparOneWall(Wall wall) {
             fyr.setComment("Vertical framing");
             fyr.setSize(height);
             fyr.setAmount(1);
+            fyr.setPrice(LogicFacade.getPrice(fyr.getId()));
```

Comment:
If you set the price here, it will be calculated for the needed size, not for the available one. I calculate de available sizes first in function layer